MessageTerminal/MessageTerminal.cxx

```diff
@@ -86,95 +86,42 @@ void move(istringstream& iss)
 
 void get_status(istringstream& iss)
 {
-    SerialResult res = comm.get_status();
+    Status status;
+    SerialResult res = comm.get_status(&status, RECV_MSG_TIMEOUT);
     if (res == SERIAL_OK) {
-        cout << "OK" << endl;
+        switch (status) {
+            case STATUS_IDLE:   puts("Status: IDLE"); break;
+            case STATUS_MOVING: puts("Status: MOVING"); break;
+            case STATUS_HOMING: puts("Status: HOMING"); break;
+            case STATUS_FAULT:  puts("Status: FAULT"); break;
+            default:            puts("ERR: Invalid status"); break;
+        }
     }
     else {
         printf("ERROR: %d\n", res);
-        return;
     }
+}
 
-    res = comm.recv_message(RECV_MSG_TIMEOUT);
+void get_position(istringstream& iss)
+{
+    Position position;
+    SerialResult res = comm.get_position(&position, RECV_MSG_TIMEOUT);
     if (res == SERIAL_OK) {
-        if (comm.received_message().id == MSG_ID_STATUS) {
-            Status status = (Status)((comm.received_message().data)[0]);
-            switch (status) {
-                case STATUS_IDLE:   puts("Status: IDLE"); break;
-                case STATUS_MOVING: puts("Status: MOVING"); break;
-                case STATUS_HOMING: puts("Status: HOMING"); break;
-                default:            puts("ERR: Invalid status"); break;
-            }
-        }
-        else {
-            printf("ERROR: Wrong ID: %d\n", comm.received_message().id);
-        }
+        printf("Position: (%d, %d)\n", position.x, position.y);
```

Comment:
Is this position in counts? Would you mind adding a comment :)?

Reply:
Good point, I'll change the variable name and print string